lib/inputs/rpm/static.ts

```diff
@@ -0,0 +1,88 @@
+import { unlink, writeFile } from "fs";
+import { EOL, tmpdir } from "os";
+import { resolve as resolvePath } from "path";
+import { getContent } from "../../extractor";
+import { ExtractAction, ExtractedLayers } from "../../extractor/types";
+import { streamToBuffer } from "../../stream-utils";
+import { CmdOutput, execute } from "../../sub-process";
+import { RpmDbPackage } from "./types";
+
+export const getRpmDbFileContentAction: ExtractAction = {
+  actionName: "rpm-db",
+  fileNamePattern: "/var/lib/rpm/Packages",
+  callback: streamToBuffer,
+};
+
+export async function getRpmDbFileContent(
+  extractedLayers: ExtractedLayers,
+): Promise<string> {
+  const apkDb = getContent(extractedLayers, getRpmDbFileContentAction);
+  if (!apkDb) {
+    return "";
+  }
+
+  const filePath = generateTempFileName();
+  await writeToFile(filePath, apkDb as Buffer);
+
+  try {
+    const cmdOutput = await execute("rpmdb", ["-f", filePath]).catch(
+      handleError,
+    );
+    return formatRpmDbOutput(cmdOutput);
+  } finally {
+    await removeFile(filePath);
+  }
+}
+
+function handleError(error): CmdOutput | never {
+  const stderr = error.stderr;
+  if (typeof stderr === "string" && stderr.indexOf("not found") >= 0) {
+    return { stdout: "", stderr: "" };
+  }
+  throw error;
+}
+
+function generateTempFileName(): string {
+  const tmpPath = tmpdir();
+  const randomFileName = Math.random().toString();
```

Comment:
does it make sense for the file name to somehow indicate the image name and file we try to pull? so if we see a file on the disk that is called rhel8_rpm_db_GUID we can know what is going on there. just a point to think about, not saying we should do it

Reply:
@ivanstanev, you may use the package `tmp` and `fileSync` to create and handle temporary files.

Reply:
@shaimendel I don't think we care about the package, we only need to extract it somewhere without clashing with an existing file. If something goes wrong we're not going to leave it lying around and consuming disk space

@moshikod That's a pretty big package for something close to what we want here but in 2 lines 😄 I really don't want to add more packages unless we really have to